pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -566,6 +587,34 @@ func TestSimpleDaemonSetPodCreateErrors(t *testing.T) {
 	}
 }
 
+func TestDaemonSetPodCreateExpectationsError(t *testing.T) {
+	for _, f := range []bool{true, false} {
+		defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ScheduleDaemonSetPods, f)()
```

Comment:
move defer out of loop. defers are called at the on func return, not on block return.